src/EnergyPlus/Coils/CoilCoolingDX.cc

```diff
@@ -677,11 +677,11 @@ void CoilCoolingDX::simulate(EnergyPlusData &state,
 
     // set some reporting variables
     this->condenserInletTemperature = condInletNode.Temp;
-    this->dehumidificationMode = useAlternateMode;
+    this->dehumidificationMode = coilMode;
 
     // set condenser inlet/outlet nodes
     // once condenser inlet is connected to upstream components, will need to revisit
-    condInletNode.MassFlowRate = this->condMassFlowRate(useAlternateMode);
+    condInletNode.MassFlowRate = this->condMassFlowRate(coilMode);
```

Comment:
This one hurts my brain. Are there only 2 modes allowed?

    Real64 CoilCoolingDX::condMassFlowRate(bool const useAlternateMode)
    {
        if (useAlternateMode) {
            return this->altModeNomSpeed().RatedCondAirMassFlowRate;
        } else {
            return this->normModeNomSpeed().RatedCondAirMassFlowRate;
        }
    }

Reply:
That function looks like this now: 

```cpp
Real64 CoilCoolingDX::condMassFlowRate(HVAC::CoilMode const mode)
{
    if (mode != HVAC::CoilMode::Normal) {
        return this->altModeNomSpeed().RatedCondAirMassFlowRate;
    } else {
        return this->normModeNomSpeed().RatedCondAirMassFlowRate;
    }
}
```

Reply:
This was an example of using an integer enumeration as a boolean. Tut tut. One advantage of using real enumerations is that they don't implicitly cast to booleans (or to anything else for that matter).

Reply:
That wasn't my point. There are only 2 choices yet there are three possible modes for subcool. This comment is outside the scope of these changes and I don't even know if my comment is correct. It's just that something looks odd.

Reply:
That's why we need some followup on this in a separate PR. There's confusion between the former `hasAlternateMode ` which was only used to track if one or both alternate modes were present vs the former `useAlternateMode` and `dehumidificationMode` which indicated the current operating mode on each call. Some functions expect 3 variants, but some like `CoilCoolingDX::condMassFlowRate` only expect two. Maybe that's correct, maybe not. The use of each one will need to be examined.